src/main/scala/org/broadinstitute/dsde/firecloud/service/StatusService.scala

```diff
@@ -1,43 +1,63 @@
 package org.broadinstitute.dsde.firecloud.service
 
-import java.text.SimpleDateFormat
-import java.util.Date
-
 import akka.actor.Actor
-import spray.json._
+import akka.actor.Props
+import akka.pattern._
+import org.broadinstitute.dsde.firecloud.{Application, FireCloudException, FireCloudExceptionWithErrorReport}
+import org.broadinstitute.dsde.firecloud.dataaccess.{AgoraDAO, RawlsDAO, SearchDAO, ThurloeDAO}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.impSystemStatus
+import org.broadinstitute.dsde.firecloud.model.{SubsystemStatus, SystemStatus}
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
+import org.broadinstitute.dsde.firecloud.service.StatusService.CollectStatusInfo
+import spray.http.StatusCodes
+import spray.httpx.SprayJsonSupport
+import spray.json.DefaultJsonProtocol._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
+
+/**
+ * Created by anichols on 4/5/17.
+ */
+object StatusService {
+  def props(statusServiceConstructor: () => StatusService): Props = Props(statusServiceConstructor())
+
+  def constructor(app: Application)()(implicit executionContext: ExecutionContext): StatusService = {
+    new StatusService(app)
+  }
+
+  case class CollectStatusInfo()
+}
 
-import spray.routing._
+class StatusService (val app: Application)
+                    (implicit protected val executionContext: ExecutionContext) extends Actor with SprayJsonSupport {
 
+  override def receive = {
+    case CollectStatusInfo => collectStatusInfo() pipeTo sender
+  }
 
-import org.broadinstitute.dsde.firecloud.FireCloudConfig
+  def collectStatusInfo(): Future[PerRequestMessage] = {
+    val subsystemExceptionHandler: PartialFunction[Any, SubsystemStatus] = {
+      case fcExceptionWithError: FireCloudExceptionWithErrorReport => SubsystemStatus(false, Some(List(fcExceptionWithError.errorReport.message)))
+      case fcException: FireCloudException => SubsystemStatus(false, Some(List(fcException.toString)))
+      case e: Exception => SubsystemStatus(false, Some(List(e.getMessage)))
+      case _: Any => SubsystemStatus(false)
```

Comment:
nit: be consistent about `getMessage` vs. `toString` on the exceptions; I think you should use `fcException.getMessage` here.

Also, the last case could be:
```
case x => SubsystemStatus(false, x.toString)
```
if you wanted to populate *some* kind of message - optional, don't hold the PR for that.